### DISCUSSION TITLE
Follower sync problem

### DISCUSSION THREAD

**Discussion by ameliakubasinska** [NightscoutFoundation/xDrip#3580] [General]
Is it a global problem about follower sync today?

**Comment by fedemotta** [comment]
A workaround is setup as a Nightscout follower while the issue is still present. Not sure if this is a xDrip issue, more likely a google play services issue.

**Reply by Riveramv1** [reply]
Having issue with follower sync not receiving readings... Updated phone recently not sure if something is wrong

**Reply by Riveramv1** [reply]
Xdrip+

**Reply by Navid200** [reply]
Have you updated the master and follower both to the 2024.07.23 release of xDrip?

**Comment by fedemotta** [comment]
I can tell you, it seems a global issue, dozens of followers reported the error.

**Comment by Kustomwireless** [comment]
We also have lost follow ability both follows of my Wife's Xdrip+ have not received any readings in a few hours . Has there been any official info released!!

**Comment by Navid200** [comment]
xDrip sync can be negatively affected if something is interrupted with Google Play service on the phones.  This is something that has to do with the phone and can be verified.

xDrip sync relies on Google Cloud messaging.  If that service is down, xDrip sync will not work.  That is on Google's side, not the phone itself.  

Considering all of a sudden everyone is reporting problems, it cannot be related to the phones.  
This has happened over the years and usually resolves itself.

**Comment by dave1812** [comment]
The followers function for both myself and wife stopped working between noon and 1pm Pacific time.  I follow her (for years) and she follows me (again, for years).  tried restarting phones,clearing app data,to no avail.  never had this happen before today.

**Reply by Navid200** [reply]
This is not the first time.  
As an example, it happened in January of 2023.  
But, it is not something we have detailed records of.  For example, I cannot tell you exactly what regions were affected then or if it was global.

**Comment by fedemotta** [comment]
Do you know any google messaging status dashboard? 
I checked cloud.google, firebase.google, etc and everything is green. 

Almost 6 hours without xDrip followers.

**Reply by Navid200** [reply]
No,

If you have Nightscout, switch to it until xDrip sync recovers.

**Comment by OpossumGit** [comment]
Could FCM depreciation be an issue? FCM is Google Cloud Messaging (isn’t it) and is being shut down since yesterday 
https://firebase.google.com/docs/cloud-messaging/migrate-v1

**Reply by ireneusz-ptak** [reply]
It is, xdrip needs to switch to V1 API.

**Comment by ameliakubasinska** [comment]
Now it works again

**Comment by richardstock1** [comment]
Still not working for me, it went down last night 😔 we monitor our daughter on our phones and they both stopped getting data at the same time.

**Reply by drgrandios** [reply]
Same here, 2 follower phones got no sync since ~22:30h yesterday (CET) (a single value has been received at a ~0:46h). Now its nearly 8 hours later without updates :-(

**Reply by richardstock1** [reply]
Hope that xdrip tream will implement an update ASAP

**Reply by drgrandios** [reply]
Just subscribed to a (monthly) paid Nightscout service.... hopefully will need it only 1 month ;-)

**Comment by ameliakubasinska** [comment]
It was 5min ... its not working longer

**Reply by richardstock1** [reply]
Hopefully whatever changes need to be made are done quickly.

**Comment by ireneusz-ptak** [comment]
@jamorham @Navid200 This time it won't fix by itself... https://firebase.google.com/support/faq#deprecated-api-shutdown

**Reply by ameliakubasinska** [reply]
Co to znaczy?

**Reply by ireneusz-ptak** [reply]
Potrzebne są zmiany w kodzie xdripa - trzeba zagonić chłopaków do roboty i poczekać na nową wersję aplikacji.

**Reply by Przemko995** [reply]
Cześć, również mam problem od wczoraj około 22, ani ja ani moja żona nie dostaje danych od mastera syna Masakra 😞.  Czyli czekamy na poprawkę x dripa trzymając kciuki że się uda ?:)

**Reply by Przemko995** [reply]
Wlasnie Zaczęło działać mam nadzieję że stabilnie  Super !!!!uff :)

**Reply by ireneusz-ptak** [reply]
@jamorham @Navid200 Would the Firebase SDK upgrade be enough to make it work? It's v15 right now, the latest one is v24: https://github.com/NightscoutFoundation/xDrip/blob/bcd2868f742298a3932218fb55162e80241c4966/app/build.gradle#L284

**Reply by Przemko995** [reply]
Niestety, to był 1 odczyt i dalej bez zmian brak danych

**Reply by ireneusz-ptak** [reply]
You may randomly receive a single read, because the message rate is being seriously limited right now.

**Reply by Przemko995** [reply]
Thank you Ireneusz for support, so we are waiting for xdrip update

**Comment by jamorham** [comment]
Thanks for reporting this I am investigating.

**Comment by BabygunDani** [comment]
It's not working here. I am in Brazil.

**Reply by Navid200** [reply]
It is being fixed.  When it is, there will be a new Nightly release and we will announce it here.

**Reply by BabygunDani** [reply]
Thanks

**Reply by MaryCalum** [reply]
That's wonderful we are lost without it thanks for the update

**Reply by Navid200** [reply]
Hopefully, it will be fixed soon.
In the meantime, you can use Nightscout to follow instead.

**Reply by MaryCalum** [reply]
Yes I signed up for a month's subscription but can't get any data on my son's phone  but there is an error message saying max syncqueuesize exceeded

**Reply by Navid200** [reply]
There are groups for helping you with Nightscout issues.

If you use our setup, you can open a new thread here in these same discussions and I will help you.
https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html  

As far as xDrip is concerned, this is how you can set up xDrip as a Nightscout uploader and a Nightscout follower respectively:  
https://navid200.github.io/xDrip/docs/Nightscout-Settings.html  
https://navid200.github.io/xDrip/docs/Follow/FollowNightscout.html

**Comment by dave1812** [comment]
My wife saved me from certain brain damage or death when she monitored my bg's when i was stuck in a nursing facility for 15 months following 4 brain surgeries.  She would call the nursing staff to alert them of any extreme highs or lows, on many occasions.  A**ll thanks to the follower function.**

**Comment by dave1812** [comment]
is nightscout hard to use?  help!

**Reply by Navid200** [reply]
Nightscout needs to be set up.  
Setting up Nightscout requires either to pay a service provider a monthly fee or to spend some (1-2 hours) once to set up.  

After it is set up, setting up xDrip to upload to it or to follow it, or using it is as simple as setting up and using xDrip sync follow.

**Reply by dave1812** [reply]
My wife has spent over a half hour trying to figure it out. She's not there yet. I don't really wanna pay for a service that's going to be temporary for us. I hope to God that the powers that be fixed xDrip follower quickly.

**Reply by Navid200** [reply]
This is one option for setting up xDrip:
https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html

It costs about $0.00-$0.03 a month.

**Reply by dave1812** [reply]
Thanks, but we're not techie enough to do all that.

**Reply by drgrandios** [reply]
> Thanks, but we're not techie enough to do all that.

There are several services available that provide ready-made Nightscout Instances. I chose Nightscout Pro which is 5 Eur per month and just some clicks to start. Hopefully within this month the follower sync is working again.

**Reply by vawlk** [reply]
Don't think of it as a "temporary" service. Think of it as a backup,

**Comment by jamorham** [comment]
There is a new nightly available which is intended to resolve this issue. Please test. thanks

**Reply by vawlk** [reply]
So far, so good. Son and I have installed it, I have switched back to XD+ follower and I am getting updates.
Thanks much for your work!

**Reply by MaryCalum** [reply]
How do i access the update please

**Reply by Przemko995** [reply]
You ale the King looks that works fine ! Thank you very much you helps a lot of People!

**Reply by dbames** [reply]
The deprecation of Firebase Messaging came at a bad time for me - just as I was trying to migrate my son's xdrip master from a partially broken phone and the xdrip+ sync follow didn't work on my/wife's handsets.  I assumed I'd got something screwed up, as I had two masters operating.  We fell back to nightscout follow, but I'm pleased to say with the latest pre-release drop, all is back up and running again.

Very speedy resolution - my eternal thanks!

**Comment by dave1812** [comment]
If we have the variant as well as the standard xdrip app on our phones
which one do we update?

**Comment by richardstock1** [comment]
Yep all working here on he nightly build.. fantastic work.. thank you kindly.. if there is a donation link then please pop it up.

**Comment by dave1812** [comment]
I've tried three times to send the file from my computer to my phone and
now I get a message from Google that it's blocked for security reasons.

**Comment by dave1812** [comment]
I finally managed to get the file for my computer to my phone. Google said
it couldn't allow the APK in an email so I had to install it on Google drive

**Comment by dave1812** [comment]
I'm still wondering if there is a variant that I need to install as a
follower app

**Comment by whoiswes** [comment]
Confirmed working here with both a standard install as well as a variant.  Everything looks good.  Thanks for the quick fix @jamorham

**Comment by dave1812** [comment]
Because my wife and I each follow each other we have two copies of extrip
on each phone. In order to do that we used to install a special app for
allowing the installation of multiple copies of an app but now we are doing
the variant instead. It's been so long I can't remember where or how to do
the variant

**Comment by dave1812** [comment]
Okay it worked. Now I have to install the update on the other phone. Thank
you thank you thank you thank you so much for getting this fixed quickly.

**Reply by Corsaro33** [reply]
Scusate sto seguendo questa discussione per far funzionare xdrip per mia figlia.
Dove trovo il file modificato ?
Grazie

**Comment by antonulasevich** [comment]
Thank you for the quick fix!

**Comment by dave1812** [comment]
You know I got it working on both watches in both directions. We installed
the update for extrip but not for the variant and it worked. The biggest
problem was that on my wife's phone I could not get extra to update for
some reason. When I tried it and tried to open the app the app refused to
open. I had to delete the original extrip app and install from scratch from
the new APK that you provided today. Then it worked but on my watch I did
not have that problem. I was able to update without issue

**Comment by xdripisaac** [comment]
Fenómenos.  Muchas gracias.

**Comment by wirbel-at-vdr-portal** [comment]
Great work! Working again.

**Comment by Kustomwireless** [comment]
We are Back working in VA
Thank you for the fast response and all the Work that u do JamorHam!!

**Comment by dave1812** [comment]
Now that we've got it working we have one little tiny issue. When we go to
system status on my wife's phone we do not see my phone listed as a
follower although I can follow her okay. On my phone as a master I can see
in the system status that her phone is attached as a follower. Any ideas?

**Reply by vawlk** [reply]
I have had that problem for a long time but it always works. Not sure how to fix that though.

**Comment by BabygunDani** [comment]
Thanks so much for solving this problem. I am just a mother trying to help my 10 yo child with his diabetes. Without knowledge on technology or programing, but willing to understand it to be able to help him.  
Thanks again.

**Comment by AnibalPellino** [comment]
New update works . Thank you

**Comment by markok314** [comment]
The fix works on Pixel 7a (master and follower). Thanks for good work and fast response!

**Comment by Josesan2014** [comment]
Hello, since yesterday the master has not sent the measurements to my two followers. I see they mention a solution, but I can't identify what it is. Can someone please help me?

**Reply by Navid200** [reply]
The solution is to install this release on the master and follower:
https://github.com/NightscoutFoundation/xDrip/releases/tag/2024.07.23

**Reply by Josesan2014** [reply]
Thanks a lot!!

**Comment by fedemotta** [comment]
Hi, thank you very much for the fix! I helped to install in several phones and it is working great. 
My recommendation is: only update if you don't have Nightscout working, an update can generate some other unexpected issues. 

Do you know when the next stable release will be available?

**Comment by cokenbowls** [comment]
+1 thank you so much for this fix.  Your guys' work is very much appreciated.

**Comment by janvitos** [comment]
Hi,

Last night, I downloaded and installed the latest xDrip nightly on all of my devices, master and followers, and that fixed the xDrip Sync Follower issue.

But this morning, the sync issue came back, and the followers stopped receiving data once again. I had to revert to Nightscout Follower for the time being.

Anyone else having sync problems again after installing the new xDrip nightly? Or is yours still working?

**Reply by utechnet** [reply]
I concur, last night I installed the new nightly release on my child's and my wife and my phones which solved the xdrip sync issue overnight. However this morning xdrip sync will be missing readings for 30 minutes or more. Then I'll change it back to nightscout source and it'll get caught up to date with readings. I can then switch it back to xdrip sync and it'll continue to get a few readings before going dark again.

**Reply by vawlk** [reply]
same here. Worked all night and this morning there seems to be a missing readings here and there. Nightscout shows all of the readings.

**Comment by Kustomwireless** [comment]
We are having issues where 2 followers are not getting readings for 15 mins at a time and then it works again the master phone is a Samsung a15 and followers are A Motorola 5g the other is a OnePlus. It has happened 3 time this morning we are in the US.

**Comment by janvitos** [comment]
Hey @jamorham and @Navid200, it seems like people are still having sync issues even after updating to the new nightly. Tagging you so you are aware of the issue. Thanks!

**Reply by Navid200** [reply]
Yes,

We have been using Google Cloud messaging for years.
Google interrupted that.  
What was released yesterday was  a Nightly.  A first release to see how it goes.
We will improve it.  

But, it is really fortunate that everyone has an alternative, Nightscout.
So, while this goes through inevitable steps to be improved, users can use Nightscout as an alternative.   
<br/>  

  
---  
  
**Nightscout**  
If you don't have Nightscout, you can look at the following that helps you decide which option is best for you: https://nightscout.github.io/nightscout/wizard/  
  
If you decide to use Google Cloud Nightscout, here are the instructions:  
https://navid200.github.io/xDrip/docs/Nightscout/GoogleCloud.html  
<br/>  
  
---  
  
**If you have Nightscout**  
This explains how to set up xDrip to be a Nightscout uploader (master): https://navid200.github.io/xDrip/docs/Nightscout-Settings.html 
  
This explains how to set up xDrip to be a Nightscout follower: https://navid200.github.io/xDrip/docs/Follow/FollowNightscout.html

**Reply by janvitos** [reply]
Thanks for the input @Navid200.

Quick question: I wouldn't mind using Nightscout all the time instead of xDrip Sync Follower if getting the BG readings was quicker. Unfortunately, sometimes, there can be a ~1 minute delay before the followers get the readings from the master. Is there any way to speed this up? Would there be any way to make it as speedy as xDrip Sync Follower, which is almost instantaneous? Thanks!

**Reply by Navid200** [reply]
If your reading from Nightscout is occasionally delayed by more than 5 minutes, you can slightly increase the added delay as explained here under follow delay: https://navid200.github.io/xDrip/docs/Follow/FollowNightscout.html  
But, if the delay is only 1 minute, no, there is no way to improve that.  

I am sorry to mention it this way:  Please say goodbye to what xDrip sync used to be because the infrastructure underneath was pulled from under us by Google.
We are working on improving things.  But, I don't know how else to clarify that what we had is gone.
We may find a similar or identical alternative.  But, I cannot make any promises.  
A 1-minute delay is the least of my worries right now.  I hope you appreciate the situation we are in and can be patient.

**Reply by janvitos** [reply]
Thanks a lot for the info @Navid200. Of course I understand, and I'm sure most xDrip users will understand too.

I think the hardest part for some will be to get Nightscout setup if they don't already have it.

**Comment by dave1812** [comment]
Unfortunately for us we can't figure out the convoluted steps of installing
night scout and paying for a service which is the last thing I want to do

**Comment by dave1812** [comment]
My wife and I are two users of follower. We follow each other. Since
yesterday when the update was made to the app it's work flawlessly

**Comment by dave1812** [comment]
My wife and i's phone both periodically get the message on the extrip
screen about there being a problem with following but all the data is there
since yesterday when the fix was made to the app

**Comment by richardstock1** [comment]
So updated to the nightly build and all was well until about 2/3 hours ago where readings have become sporadic on both the follow phones (Pixel 8 and Pixel 6) with 15-30 mins gaps between readings.

**Comment by Agonyzyr95** [comment]
Thank you so much, you guys fixed this faster than any professional company would. You guys are life savers and I wish we could make you all rich like Abbott or Medtronic since you have a better program than them and better morals

**Comment by richardstock1** [comment]
Not sure if anyone is having sync issues again. For the last 2 hours we have been getting sporadic readings and big delays between the master and follow phones.

**Reply by utechnet** [reply]
@richardstock1 Things are not perfect, but at least mostly work well.  I too experience some times when the xdrip xync does not work for 15 minutes at a time or more intermittently.  I use this to monitor my child, which is obviously very important to be timely and accurate.  My wife and I both monitor our child.  I have 1 of our phones setup to monitor via xdrip sync and the other via Nightscout.  That's why it's important to have a backup method such as Nightscout for when things fail.  I keep watching the repo's activity https://github.com/NightscoutFoundation/xDrip/activity to look out for commits related to Google Cloud Messaging to improve the small glitches that we experience with it.  3 weeks have gone by and the only commit was the initial code change that was needed and is what we're working off of now.  I'm not complaining at all, this stuff doesn't cost me anything at all, and it takes many developers' knowledge, skills and time to make changes happen.  Many thanks devs!

**Comment by MaryCalum** [comment]
Yes we are having the same issue Sent from my phone-------- Original message --------From: richardstock1 ***@***.***>Date: Tue 13 Aug 2024, 7:21 p.m.To: NightscoutFoundation/xDrip ***@***.***>Cc: MaryCalum ***@***.***>, Comment ***@***.***>Subject: Re: [NightscoutFoundation/xDrip] Follower sync problem (Discussion #3580)
Not sure if anyone is having sync issues again. For the last 2 hours we have been getting sporadic readings and big delays between the master and follow phones.

**Comment by Navid200** [comment]
It helps to know if you have updated both the master and the follower to the xDrip release of 2024.07.23 or any after that or not.

**Reply by richardstock1** [reply]
Yes I updated them three weeks ago as per this thread..

**Comment by MaryCalum** [comment]
Yes we have updated both phones to the July update

**Comment by BabygunDani** [comment]
I am also having problems. Some minutes without updating the follower.

**Comment by DaNitroo** [comment]
We don't have any problems. We also use the version from 23.7. on 4 phones (master and follower) and everything works great. Thanks again for that!

**Comment by JensChr4** [comment]
After updating to the newest release, Nightly build 7th Aug 2024, the updates for followers became wonky (e.g. some came through, others didn't) and then petered out till nothing. Problem persisted on WiFi as well as over 5G. Living in Norway. Reverted to relying on Dexcom follower, which is a great feature. However, it's a bit slower and turning off alarms isn't synchronized.

**Reply by BoarderElSr** [reply]
Hello, how are you?
I would like to ask if you are aware of any synchronization issues with the follower. Since this afternoon, it has not been syncing properly, and the signal was being lost.
Is anyone else experiencing this?
Thank you all in advance.

**Reply by Navid200** [reply]
Yes, many have reported it.
We switched to a new server infrastructure about a month ago.  It will take a while to smooth out all the issues.
You can use Nightscout in the meantime.  Other than that, you will have to be patient.
And thanks for reporting.

**Reply by BoarderElSr** [reply]
Navid200, thank you for your quick response.
Yes, I remember the event in July when it wasn't syncing, and resolved it very quickly. However, starting this afternoon, the issue has reappeared.

**Reply by Navid200** [reply]
And when I stated "Yes, many have reported it",  I meant that many have reported issues today.  So, we are aware that there are issues now.
And there is nothing we can do to fix it right now.

  xDrip sync is not the only way to follow in xDrip.  There is a workaround, Nightscout.

**Reply by BoarderElSr** [reply]
Navid200, Thank you for the clarification and for keeping me informed. I appreciate your efforts in addressing the issue. Thanks again for your support.

**Comment by dave1812** [comment]
Same here. I can't figure out why people are having trouble recently. My
wife and I follow each other and it works 100% of the time day in and day
out. Of course we had to get the update a few weeks ago but that fixed it
for us

**Comment by dave1812** [comment]
We only updated our two Masters. Been working fine ever since. We did not
touch the installation of the two follower apps

**Comment by MaryCalum** [comment]
We updated master and follower phones on the 22rd of July  thanks Sent from my phone-------- Original message --------From: Dave A ***@***.***>Date: Wed 14 Aug 2024, 4:11 p.m.To: NightscoutFoundation/xDrip ***@***.***>Cc: MaryCalum ***@***.***>, Comment ***@***.***>Subject: Re: [NightscoutFoundation/xDrip] Follower sync problem (Discussion #3580)
We only updated our two Masters. Been working fine ever since. We did not

touch the installation of the two follower apps

**Comment by Riveramv1** [comment]
Hi I deleted app, reinstalled all and worked. Thank you so much. Virginia

Yahoo Mail: Search, organise, conquer 
 
  On Fri, 6 Sept 2024 at 12:00, ***@***.***> wrote:   


Have you updated the master and follower both to the 2024.07.23 release of xDrip?

**Comment by vawlk** [comment]
seems syncing isn't working again today. Started this morning around 11am CDT and been on and off all day while nightscout is showing all data.
